src/configuration/parse.ts

```diff
@@ -0,0 +1,237 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as path from "path";
+import * as toAbsoluteGlob from "to-absolute-glob";
+import { IOptions, RuleSeverity } from "../index";
+import { arrayify, hasOwnProperty } from "../utils";
+import { IConfigurationFile } from "./configurationFile";
+import {
+    getRulesDirectories,
+    RawConfigFile,
+    RawRuleConfig,
+    RawRulesConfig,
+    resolveConfigurationPath,
+} from "./read";
+
+/**
```

Comment:
`parseRuleOptions` used to return an object that always had a `ruleArguments` and `ruleSeverity` key. Now it returns a compact object so that merging is simpler later. This modified behavior can be seen in the "parses different ways of storing options" test.